src/librustc_mir/borrow_check/mod.rs

```diff
@@ -830,6 +835,37 @@ impl InitializationRequiringAction {
     }
 }
 
+/// A simple linked-list threaded up the stack of recursive calls in `visit_terminator_drop`.
+#[derive(Debug)]
+struct SeenTy<'a, 'gcx: 'a>(ty::Ty<'gcx>, Option<&'a SeenTy<'a, 'gcx>>);
```

Comment:
You can rearrange this to have `Option` around a pair of `Ty` and `&SeenTy`, bypassing the problems needing `Option` around it later.

Reply:
Okay I will try that.